micrometer-core/src/test/java/io/micrometer/core/instrument/binder/logging/Log4j2MetricsTest.java

```diff
@@ -368,4 +368,49 @@ void bindingTwiceToSameRegistry_doesNotDoubleCount() {
         assertThat(registry.get("log4j2.events").tags("level", "error").counter().count()).isEqualTo(2);
     }
 
+    @Issue("#5901")
+    @Test
+    void programmaticallyAddedLoggerConfigShouldBeCounted() {
+        LoggerContext context = new LoggerContext("test");
+
+        AtomicInteger newLoggerUpdatedWithLoggerFilter = new AtomicInteger();
+
+        // checks that updateLoggers() is called after MetricsFilter is added to the
+        // programmatically added logger
+        context.addPropertyChangeListener(event -> {
+            if (event.getNewValue() instanceof Configuration) {
+                Configuration newConfiguration = (Configuration) event.getNewValue();
+                LoggerConfig loggerConfig = newConfiguration.getLoggerConfig("com.test");
+                if (!loggerConfig.isAdditive() && loggerConfig.getFilter() instanceof Log4j2Metrics.MetricsFilter) {
+                    newLoggerUpdatedWithLoggerFilter.incrementAndGet();
```

Comment:
I wonder if we should assert on this. The only effect we need to ensure is that log events for programmatically added loggers are counted after calling `updateLoggers`, unless I'm missing something. It may be a bit too into the details of the implementation to check this. What do you think?

Reply:
Sure, I was considering whether to add this check or not. I added it because it is the only thing verifying `updateLoggers()` is called at all (at that it is once).

But I agree that its a bit too into the details. I'll remove it

Reply:
We could also remove the configured logger filters checks, as that is also a bit into the details, and replace them with just checking that:
1. It is not counted when added to `Configuration` but `updateLoggers()` is not yet called
2. It is counted after `updateLoggers()` is called
3. It is no longer counted when the `Log4j2Metrics` is closed